db/mongodb.go

```diff
@@ -0,0 +1,192 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"time"
+)
+
+var (
+	cronCollection     = "cronjobs"
+	mongoCollectionMap = map[string]string{
+		"id":       "_id",
+		"IsActive": "active",
+		"Function": "function",
+		"Workload": "workload",
+		"CronTime": "time",
+		"TimeZone": "tz",
+	}
+)
+
+type ErrUnknownField struct {
+	Field string
+}
+
+type ErrUnsupportedWorkloadType struct {
+	Type string
+}
+
+func (e ErrUnknownField) Error() string {
+	return fmt.Sprintf("Unknown cron field: %s", e.Field)
+}
+
+func (e ErrUnsupportedWorkloadType) Error() string {
+	return fmt.Sprintf("Unsupported workload type: %s", e.Type)
+}
+
+type MongoCronJob struct {
```

Comment:
I was thinking about that, but then CronJob becomes DB specific. Also, the workload for `MongoCronJob` can be an interface (so it can be either a `bson.M` object or a string), while from my perspective `CronJob` might want to just think in strings. For example, I'm not sure SQL DBs will naturally want to think of Workloads as json like objects.